test/sqllogictest/list.slt

```diff
@@ -45,6 +45,21 @@ select LIST[1, null] :: INT LIST
 ----
 {1,NULL}
 
+query T
+select LIST[[1, null], []] :: INT LIST LIST
+----
+{{1,NULL},{}}
+
+# Lists support arbitrarily deep nesting
+query T
+SELECT LIST[[[[1], [2]]], [[[3]]]]
+----
+{{{{1},{2}}},{{{3}}}}
+
+# List(Int) cannot be cast to List(List(Int))
+query error Cannot create list with mixed types
+select LIST[1, null] :: INT LIST LIST
```

Comment:
Totally fine to leave this as is because you're just following the existing style in this file, which is 👍, but the idiomatic formatting for this would be `SELECT LIST[1, NULL]::int list list`.